account.c

```diff
@@ -48,7 +48,45 @@ void account_free(struct Account **a)
     return;
 
   if ((*a)->adata)
-    (*a)->free_adata((void **) a);
+    (*a)->free_adata(&(*a)->adata);
 
   FREE(a);
 }
+
+/**
+ * account_remove - Remove a Account from AllAccounts
+ * @param a Account to remove
+ */
+void account_remove(struct Account *a)
+{
+    struct Account *np = NULL;
+    struct Account *nptmp = NULL;
+    TAILQ_FOREACH_SAFE(np, &AllAccounts, entries, nptmp)
```

Comment:
No need to use the safe version, as you're breaking out of the loop after having invalidated the iterator.